spec/classes/collectd_plugin_mongodb_spec.rb

```diff
@@ -0,0 +1,206 @@
+      end
+    end
+
+    context 'set to an invalid value' do
+      let :params do
+        default_params.merge(:interval => 'invalid',)
+      end
+
+      it 'should fail' do
+        expect do
+          should contain_class('collectd::plugin::mongodb')
+        end.to raise_error(Puppet::Error, /Expected first argument to be a Numeric or Array, got String/)
+      end
+    end
+  end
+
+  describe 'with db_host parameter' do
+    context 'set to a valid IP' do
+      let :params do
+        default_params.merge(:db_host => '127.0.0.1',)
+      end
+
+      hostdb_fixture = File.read(fixtures('plugins/mongodb.conf.hostdb'))
+      it { should contain_file('mongodb.load').with_content(hostdb_fixture) }
+    end
+
+    %w('127001', nil).each do |value|
+      context 'set to and invalid IP' do
+        let :params do
+          default_params.merge(:db_host => "#{value}",)
+        end
+
+        it 'should fail' do
+          expect do
+            should contain_class('collectd::plugin::mongodb')
+          end.to raise_error(Puppet::Error, /must be a valid IP address/)
+        end
+      end
+    end
+  end
+
+  describe 'with db_user parameter' do
+    context 'set to a valid user name' do
+      let :params do
+        default_params.merge(:db_user => 'test_user',)
+      end
+
+      confdb_fixture = File.read(fixtures('plugins/mongodb.conf.db_user'))
+      it { should contain_file('mongodb.load').with_content(confdb_fixture) }
+    end
+
+    context 'undefined' do
+      let :params do
+        { :db_pass => 'password',
+        }
+      end
+
+      it 'should fail' do
+        expect do
+          should contain_class('collectd::plugin::mongodb')
+        end.to raise_error(Puppet::Error, /is <undef> and must be a mongodb username/)
+      end
+    end
+  end
+
+  describe 'with db_pass parameter' do
+    context 'set to a valid password' do
+      let :params do
+        default_params.merge(:db_pass => 'foo',)
+      end
+
+      dbpass_fixture = File.read(fixtures('plugins/mongodb.conf.db_pass'))
+      it { should contain_file('mongodb.load').with_content(dbpass_fixture) }
+    end
+
+    context 'undefined' do
+      let :params do
+        { :db_user => 'test_user',
+        }
+      end
+
+      it 'should fail' do
+        expect do
+          should contain_class('collectd::plugin::mongodb')
+        end.to raise_error(Puppet::Error, /collectd::plugin::mongodb::db_pass is <undef>/)
+      end
+    end
+  end
+
+  describe 'with configured_dbs parameter' do
+    context 'set to a valid value with db_port defined and a single db' do
+      let :params do
+        default_params.merge(:configured_dbs => [25],
+                             :db_port        => '8080',)
+      end
+
+      dbport_single_fixture = File.read(fixtures('plugins/mongodb.conf.configured_dbs_single'))
+      it { should contain_file('mongodb.load').with_content(dbport_single_fixture) }
+    end
+  end
+
+  describe 'with configured_dbs parameter' do
```

Comment:
Style: this `describe` block shares the name with the previous one, I believe?

The `context` blocks could just as well share one `describe` block.